### PR TITLE
Chore UI

### PR DESCRIPTION
**Pull request by athibau2**
Functionality for integrating the ChoreProvider into the Child home page complete. Chore list by id is retrieved on page load. Additionally, the chore details dialog is complete. Lastly, creating a new chore via the new chore dialog is complete and functional.

### CODE REVIEW

File: togetherly/assets/images/logo-name.png
Comment (NWalker1208) [inline togetherly/assets/images/logo-name.png:1]: Why the change? In my opinion, the black outline looks worse.

File: togetherly/lib/services/chore_service.dart
Comment (NWalker1208) [inline togetherly/lib/services/chore_service.dart:1]: Have you tested that saving the due date in this way works as expected? The database column has time zone information, but I don't think Dart's DateTime does (it can process a time zone during parsing, but only to localize the time, as far as I know).

File: togetherly/lib/views/screens/child_home.dart [hunk 1]
```diff
@@ -11,27 +16,63 @@ class ChildHomePage extends StatefulWidget {
...
-            SizedBox(height: 20.0),
-            ChoreList(
-              title: 'Upcoming',
-            ),
-            SizedBox(height: 20.0),
-            ChoreList(
-              title: 'Overdue',
-            ),
-          ],
+    return Consumer<ChoreProvider>(
```
Comment (NWalker1208) [inline togetherly/lib/views/screens/child_home.dart]: Does the consumer need to cover the entire page? Or could it be moved down the hierarchy to wrap just the chore list widget? Or taking that further, could it be moved inside the chore list widget? The lower in the hierarchy the better, since every state change will trigger a rebuild of the tree beneath the consumer widget.

File: togetherly/lib/views/widgets/chore_list.dart [hunk 2]
```diff
@@ -49,10 +25,9 @@ class _ChoreListState extends State<ChoreList> {
           style: AppTextStyles.brandAccentLarge,
         ),
         Column(
-          children: <Widget>[
-            for (final chore in chores)
-              ChoreItem(chore: chore),
-          ],
+          children: widget.choreList.map((chore) {
+            return ChoreItem(chore: chore,);
+          }).toList(),
```
Comment (NWalker1208) [inline togetherly/lib/views/widgets/chore_list.dart]: Why the change here? I feel that the former syntax was more readable.

File: togetherly/lib/views/widgets/chore_list.dart [hunk 3]
```diff
@@ -4,39 +4,15 @@ import 'package:togetherly/themes.dart';
...
-    ),
-    Chore(
-      title: 'Walk the dog',
-      dueDate: DateTime(2024, 2, 14),
-      points: 20,
-      status: ChoreStatus.completed,
-      assignedChildId: 0,
-      isShared: false,
-    ),
-  ];
```
Comment (NWalker1208) [inline togetherly/lib/views/widgets/chore_list.dart:39]: Will this widget need to maintain state in the future? Or can it be made stateless?

File: togetherly/lib/views/widgets/chore_item.dart [hunk 4]
```diff
@@ -18,7 +19,8 @@ class ChoreItem extends StatelessWidget {
     }
 
     bool isDueToday(Chore chore) {
-      return chore.dueDate == getDateToday();
+      return chore.dueDate.toString().split(' ')[0] ==
+          getDateToday().toString();
```
Comment (NWalker1208) [inline togetherly/lib/views/widgets/chore_item.dart]: Would it not be more readable and robust to define a helper method that takes two DateTime objects and compares `a.year == b.year && a.month == b.month && a.day == b.day`, rather than compare strings in this way?

File: togetherly/lib/views/widgets/chore_details_dialog.dart [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+          style: AppTextStyles.brandAccentLarge,
+        ),
+      ],
+    );
+  }
+
+  String prettyDate(String date) {
+    List<String> dateTimeArr = date.split(' ');
+    List<String> dateArr = dateTimeArr.elementAt(0).split('-');
+    return '${dateArr.elementAt(1)}/${dateArr.elementAt(2)}/${dateArr.elementAt(0)}';
```
Comment (NWalker1208) [inline togetherly/lib/views/widgets/chore_details_dialog.dart]: This could be more readable if you instead took a DateTime object and pulled out `date.month`, `date.day`, and `date.year` into variables, rather than splitting the string representation.

File: togetherly/lib/views/widgets/new_chore_dialog.dart [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+        title: _title,
+        description: _description,
+        points: _points,
+        dueDate: _dueDate,
+        isShared: _isShared,
+        assignedChildId: 1,
+      );
+      setState(() => _loading = true);
+      provider.addChore(newChore);
+      setState(() => _loading = false);
```
Comment (NWalker1208) [inline togetherly/lib/views/widgets/new_chore_dialog.dart:77]: This should not occur until the future returned by addChore has returned. You could make this function async, or you could use the `Future.then` method ([see docs](https://dart.dev/guides/libraries/futures-error-handling#the-future-api-and-callbacks)).

File: togetherly/lib/views/widgets/new_chore_dialog.dart [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+      }
+    }
+    return null;
+  }
+
+  String prettyDate(String date) {
+    List<String> dateTimeArr = date.split(' ');
+    List<String> dateArr = dateTimeArr.elementAt(0).split('-');
+    return '${dateArr.elementAt(1)}/${dateArr.elementAt(2)}/${dateArr.elementAt(0)}';
+  }
```
Comment (NWalker1208) [inline togetherly/lib/views/widgets/new_chore_dialog.dart:99]: I saw this method elsewhere and left a comment on it. If you're using it in multiple places, it may be a good candidate to move into the `utility` folder.

File: togetherly/lib/views/widgets/new_chore_dialog.dart [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
+                  validator: (value) {
+                    if (int.tryParse(value!) != null) {
+                      if (int.tryParse(value)! > 100) {
+                        return 'Value cannot be larger than 100';
+                      }
+                    } else {
+                      return 'Value can only contain numeric characters';
+                    }
+                    return null;
+                  },
```
Comment (NWalker1208) [inline togetherly/lib/views/widgets/new_chore_dialog.dart]: It seems like there may be a better way to do this kind of validation: https://stackoverflow.com/questions/49577781/how-to-create-number-input-field-in-flutter

**Review by NWalker1208 — CHANGES_REQUESTED** [review]
See in-line comments.

File: togetherly/assets/images/logo-name.png
Comment (athibau2) [reply togetherly/assets/images/logo-name.png:1]: The white outline made the logo seem washed out and blobby. The black outline helps define the shapes.

File: togetherly/lib/services/chore_service.dart
Comment (athibau2) [reply togetherly/lib/services/chore_service.dart:1]: Yes, I have tested it. When defining the dueDate, DateTime returns `yyyy-mm-dd 00:00:00` for the selected date, which is the same format as the database timestamp. Currently, the database datedue is of type timestamp without timezone, so that's irrelevant. And, it resolves the format to a string anyways, so I had to perform the `.toString()` in the service because the insertion would fail otherwise since it was trying to insert a Dart DateTime as a Postgres timestamp (i.e. a string).

File: togetherly/lib/views/screens/child_home.dart (hunk 1 above)
Comment (athibau2) [reply togetherly/lib/views/screens/child_home.dart]: That's a fair point, I could move the consumer to inside the chore list widget. I'll make that change.

File: togetherly/lib/views/widgets/chore_list.dart (hunk 2 above)
Comment (athibau2) [reply togetherly/lib/views/widgets/chore_list.dart]: It's more concise and utilizes more of Darts list manipulation methods. And it's more consistent. Using a map with toList() is used more frequently throughout the app than a for loop, including in the service. I only have one remaining for loop, which I'm going to replace with a map method.

File: togetherly/lib/views/widgets/chore_list.dart (hunk 3 above)
Comment (athibau2) [reply togetherly/lib/views/widgets/chore_list.dart:39]: I assumed that any widget consuming a provider would need to be stateful, but looking at that now I see that the widget containing the AppProviders consumer is stateless, so I can change this one to be stateless as well.

File: togetherly/lib/views/widgets/chore_item.dart (hunk 4 above)
Comment (athibau2) [reply togetherly/lib/views/widgets/chore_item.dart]: Compared to the longer statement of the toString().split(' ')[0], yes it may be more readable, but overall I disagree. I find it tedious to compare individual parts of a single object rather than the entire thing. If we later need more separation between what parts of the date we are comparing, then your suggestion would definitely make more sense, but for the moment I think this makes more sense.

File: togetherly/lib/views/widgets/chore_details_dialog.dart (hunk 5 above)
Comment (athibau2) [reply togetherly/lib/views/widgets/chore_details_dialog.dart]: That's much better, I'll do that.

File: togetherly/lib/views/widgets/new_chore_dialog.dart (hunk 7 above)
Comment (athibau2) [reply togetherly/lib/views/widgets/new_chore_dialog.dart:99]: Good point.

File: togetherly/lib/views/widgets/new_chore_dialog.dart (hunk 6 above)
Comment (athibau2) [reply togetherly/lib/views/widgets/new_chore_dialog.dart:77]: Will fix

File: togetherly/lib/views/widgets/new_chore_dialog.dart (hunk 8 above)
Comment (athibau2) [reply togetherly/lib/views/widgets/new_chore_dialog.dart]: Thanks for the tip, I added the inputFormatters line which eliminates the need for the outer if statement in the validator.

File: togetherly/lib/utilities/date.dart [hunk 9]
```diff
@@ -0,0 +1,14 @@
+class DateHelpers {
+  DateTime getDateToday() {
+    DateTime date = DateTime.now();
+    return DateTime(date.year, date.month, date.day);
+  }
+
+  bool isDueToday(DateTime dueDate) {
+    return dueDate.toString().split(' ')[0] == getDateToday().toString();
```
Comment (NWalker1208) [inline togetherly/lib/utilities/date.dart]: I still think that doing this without strings will be more readable and robust. At the moment, the logic here simply doesn't work, as is evidenced by running the following code in DartPad:
```dart

void main() {
  DateTime now = DateTime.now();
  DateTime today = DateTime(now.year, now.month, now.day);
  DateTime dueDate = today;
  
  print(dueDate.toString().split(' ')[0]);
  print(today.toString());
  
  print(dueDate.toString().split(' ')[0] == today.toString());
}
```
It's not a big deal though, so I'll push some changes that make this more concise, readable, and robust.

File: togetherly/lib/services/chore_service.dart
Comment (NWalker1208) [reply togetherly/lib/services/chore_service.dart:1]: In my last minute fixes on Friday, I had changed the column type to be a timestamp with a time zone, since Supabase discourages using timestamps without time zones. I guess it got changed back. We can leave it as is since the dueDate stuff is only temporary anyways (as we'll be implementing the ChoreAssignment system).

File: togetherly/lib/views/widgets/chore_item.dart (hunk 4 above)
Comment (NWalker1208) [reply togetherly/lib/views/widgets/chore_item.dart]: I will address this in some changes that I push.

File: togetherly/assets/images/logo-name.png
Comment (NWalker1208) [reply togetherly/assets/images/logo-name.png:1]: image
This is how it renders for me. The black outline causes some aliasing issues. Perhaps a drop shadow would work better?

Comment (athibau2) [reply togetherly/assets/images/logo-name.png:1]: I pushed a drop shadow logo. Try it out and see if you like it.

Comment (NWalker1208) [reply togetherly/assets/images/logo-name.png:1]: Looks great now, thanks!

**Review by NWalker1208 — APPROVED** [review]
